examples/credentials.py

```diff
@@ -0,0 +1,28 @@
+import json
+from typing import TypedDict
+
+
+class Credentials(TypedDict):
+    username: str
+    password: str
+
+
+def get_credentials_for_domain(url: str | None) -> tuple[str | None, str | None]:
+    """Get credentials for a given domain from environment variables."""
+    if not url:
+        return None, None
+
+    import os
+    from urllib.parse import urlparse
```

Comment:
It's a bad practice in python to use imports inside function body. Move them on top of file (or even better use the utils file in `utils/url.py`) to parse the url